extensions/amp-mowplayer/0.1/amp-mowplayer.js

```diff
@@ -223,17 +219,28 @@ class AmpMowplayer extends AMP.BaseElement {
       return; // We only process valid JSON.
     }
 
-    const eventType = data['event'];
-    const info = data['info'] || {};
+    if (data['mowplayer'] === undefined) {
+      return;
+    }
+
+    const eventType = data['mowplayer']['type'];
+    const info = getData(data['mowplayer']) || {};
 
     const {element} = this;
 
-    if (eventType === 'set_aspect_ratio') {
-      this.attemptChangeHeight(info['new_height']).catch(() => {});
+    if (eventType === 'handshake') {
+      this.sendMessage_('handshake_done', {});
+    } else if (eventType === 'visibility_observer') {
+      this.onVisibilityObserver_();
+    } else if (eventType === 'pause') {
+      this.pause();
+    } else if (eventType === 'play') {
```

Comment:
**Don't implement playback controlled from the `iframe` to the document.**

This already registers to the `VideoManager`. You don't need to control autoplay like this, you can simply add an `autoplay` attribute and restore the playback method implementations you removed below.